File: src/main/java/com/keisu/redditclone/models/Post.java
package com.keisu.redditclone.models;


import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Post {
    @Id
    @GeneratedValue
    private long id;
    private String link;
    private String user;

    public Post(){}

    public Post(long id, String link, String user) {
        this.id = id;
        this.link = link;
        this.user = user;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getId() {
        return id;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getLink() {
        return link;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }
}

File: src/main/webapp/app/ts/models/post.ts
export class Post {
    private link:string;
    private user:string;

    public set setLink(link:string) {
        this.link = link;
    }

    public get getLink():string {
        return this.link;
    }

    public set setUser(user:string) {
        this.user = user;
    }

    public get getUser(): string {
        return this.user;
    }
}